File: new/frontend/src/api/history.ts
import apiClient from './client'
import type { Task } from '@/types'

export interface HistoryTaskListResponse {
  success: boolean
  tasks: any[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface HistoryTaskDetailResponse {
  success: boolean
  metadata: any
  storyboard?: any
}

/** List tasks */
export function listTasks(params?: { status?: string; limit?: number }) {
  return apiClient.get('/history/tasks', {
    params: {
      status: params?.status || undefined,
      page: 1,
      page_size: params?.limit || 200,
    },
  }) as unknown as Promise<HistoryTaskListResponse>
}

/** Get task detail */
export function getTask(taskId: string) {
  return apiClient.get(`/history/tasks/${taskId}`) as unknown as Promise<HistoryTaskDetailResponse>
}

/** Cancel task */
export function cancelTask(taskId: string) {
  return apiClient.delete(`/history/tasks/${taskId}`) as unknown as Promise<{ success: boolean; message: string }>
}

File: new/backend/schemas/history.py
from typing import Optional, List, Any
from pydantic import BaseModel, Field


class TaskListQuery(BaseModel):
    page: int = Field(1, ge=1)
    page_size: int = Field(20, ge=1, le=100)
    status: Optional[str] = None
    sort_by: str = "created_at"
    sort_order: str = "desc"


class TaskMetadataBrief(BaseModel):
    task_id: str
    title: Optional[str] = None
    status: Optional[str] = None
    created_at: Optional[str] = None
    completed_at: Optional[str] = None
    duration: Optional[float] = None
    file_size: Optional[int] = None


class TaskListResponse(BaseModel):
    success: bool = True
    tasks: List[Any] = []
    total: int = 0
    page: int = 1
    page_size: int = 20
    total_pages: int = 0


class TaskDetailResponse(BaseModel):
    success: bool = True
    metadata: Optional[Any] = None
    storyboard: Optional[Any] = None


class StatisticsResponse(BaseModel):
    success: bool = True
    total_tasks: int = 0
    completed: int = 0
    failed: int = 0
    total_duration: float = 0.0
    total_size: int = 0


class DuplicateResponse(BaseModel):
    success: bool = True
    message: str = "Success"
    data: Optional[Any] = None


class DeleteResponse(BaseModel):
    success: bool = True
    message: str = "Task deleted"
